src/server.h

```diff
@@ -401,6 +401,7 @@ typedef enum blocking_type {
     BLOCKED_ZSET,    /* BZPOP et al. */
     BLOCKED_POSTPONE, /* Blocked by processCommand, re-try processing later. */
     BLOCKED_SHUTDOWN, /* SHUTDOWN. */
+    BLOCKED_WAIT_AFTER_REPL, /* Blocked until replication is complete. */
```

Comment:
can't we either use BLOCKED_WAIT for this purpose, or even abandon the implicit blocking and instruct the client (redis-cli) to issue a WAIT command?

Reply:
removing this will also revert all the complications you added around CLIENT_PENDING_COMMAND in replication.c and blocked.c (will actually leave these two files unmodified)

Reply:
I think this is the same question @madolson raised in this [comment](https://github.com/redis/redis/pull/10517/files#r1102121723). I have provided a response [here](https://github.com/redis/redis/pull/10517/files#r1106868410). Please take a look.

Reply:
resolving since the other [thread](https://github.com/redis/redis/pull/10517/files#r1102121723) has more context. Let's continue the discussion there.